### PR TITLE
final doc updates

### LINKED ISSUE #281: Docs for 0.2
Trying to make a list of updates since 0.1.1 that need to be reflected in docs. [Judging by this from around first release](https://github.com/geopandas/geopandas/compare/a4a88c7e96c758f2d50e7af907b...master), I'm seeing:

* update requirements (and add installation instructions for each)
* Add `sjoin`
* New options for `plot`
* New options for `to_file`
* Add `iterfeatures`

Missing anything?

I may also re-organize the functions section (if people don't mind) into:

* Input / output
* Managing Projections
* Geometric manipulations
* Mapping

Also: do we have any spatial data loaded in the library a user can just import? Helpful for examples.

### PR DESCRIPTION
**Pull request by nickeubank**
Closes  #281?

- Updates tool import paths -- `sjoin` and `overlay` now in root path. 
- Adds dissolve docs

### CODE REVIEW

File: doc/source/aggregation_with_dissolve.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
+.. ipython:: python
+   :suppress:
+
+   import geopandas as gpd
+
+
+Aggregation with `dissolve`
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: Can you use double backticks for code?

File: doc/source/aggregation_with_dissolve.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+`dissolve` can be thought of as doing three things: (a) it dissolves all the geometries within a given group together into a single geometric feature (using the `unary_union` method), and (b) it aggregates all the rows of data in a group using `groupby.aggregate()`, and (c) it combines those two results.
+
+`dissolve` Example
+~~~~~~~~~~~~~~~~~~
+
+Suppose we are interested in studying continents, but we only have country-level data like the country dataset included in *geopandas*. We can easily convert this to a continent-level dataset.
+
+
+First, let's look at the most simple case where we just want continent shapes and names. By default, `dissolve` will pass `'first'` to `groupby.aggregate`, which can be very useful for getting the names of an aggregated shape.
```
Comment (jorisvandenbossche) [inline doc/source/aggregation_with_dissolve.rst]: "which can be very useful for getting the names of an aggregated shape" -> this is not fully correct I think? Or at least not the case in this example, as there are no columns to aggregate

File: doc/source/geometric_manipulations.rst [hunk 3]
```diff
@@ -40,6 +40,11 @@ Constructive Methods
   Returns a ``GeoSeries`` containing a simplified representation of
   each object.
 
+.. attribute:: GeoSeries.unary_union
+
+Return a geometry containing the union of all geometries in the ``GeoSeries``.
```
Comment (jorisvandenbossche) [inline doc/source/geometric_manipulations.rst]: indentation missing

File: doc/source/aggregation_with_dissolve.rst (hunk 2 above)
Comment (nickeubank) [inline doc/source/aggregation_with_dissolve.rst]: yeah -- problem with example data -- no extra string variables. It comes up a lot in practice when one has variables that are already at the level to which one is aggregating. For example, if one is aggregating states to countries and there's a column for country leader's name, or country code, etc. But wasn't sure how to deal with given existing example data. I'll just drop it.

**Pull request merged**